File: restore_files.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const filesToRestore = [
    {
        sha: "b28711e74a87a221f75960098df29ea9741dc5f9",
        targetPath: "src/app/modules/ModuleList.tsx"
    },
    {
        sha: "be6814fa89a42f89a42f89a42f89a42f89a42f89",
        targetPath: "src/app/dashboard/page.tsx"
    },
    {
        sha: "16ec6ef6f5e5694c0072c7263c959451121d556c",
        targetPath: "src/app/dashboard/layout.tsx"
    }
];

function restore() {
    // Ensure directories exist
    if (!fs.existsSync("src/app/dashboard")) fs.mkdirSync("src/app/dashboard", { recursive: true });
    if (!fs.existsSync("src/app/modules")) fs.mkdirSync("src/app/modules", { recursive: true });
    
    filesToRestore.forEach(item => {
        console.log(`Restoring ${item.targetPath} using blob ${item.sha}...`);
        try {
            const content = execSync(`git cat-file -p ${item.sha}`);
            fs.writeFileSync(item.targetPath, content);
            console.log(`Successfully restored ${item.targetPath}`);
        } catch (error) {
            console.error(`Failed to restore ${item.targetPath}: ${error.message}`);
        }
    });
}

restore();

File: restore_files.py
import subprocess
import os

files_to_restore = [
    {
        "sha": "b28711e74a87a221f75960098df29ea9741dc5f9",
        "path": "src/app/modules/ModuleList.tsx"
    },
    {
        "sha": "be6814fa89a42f89a42f89a42f89a42f89a42f89",
        "path": "src/app/dashboard/page.tsx"
    },
    {
        "sha": "16ec6ef6f5e5694c0072c7263c959451121d556c",
        "path": "src/app/dashboard/layout.tsx"
    }
]

def restore():
    # Ensure directories exist
    os.makedirs("src/app/dashboard", exist_ok=True)
    os.makedirs("src/app/modules", exist_ok=True)
    
    for item in files_to_restore:
        sha = item["sha"]
        path = item["path"]
        print(f"Restoring {path} using blob {sha}...")
        try:
            content = subprocess.check_output(["git", "cat-file", "-p", sha])
            with open(path, "wb") as f:
                f.write(content)
            print(f"Successfully restored {path}")
        except Exception as e:
            print(f"Failed to restore {path}: {e}")

if __name__ == "__main__":
    restore()
